Source/Core/PixelFormat.js

```diff
@@ -281,6 +282,41 @@ define([
                 componentsLength = 1;
             }
             return componentsLength * PixelDatatype.sizeInBytes(pixelDatatype) * width * height;
+        },
+
+        /**
+         * @private
+         */
+        createTypedArray : function(pixelFormat, pixelDatatype, width, height) {
+            var constructor;
+            var sizeInBytes = PixelDatatype.sizeInBytes(pixelDatatype);
+            if (sizeInBytes === Uint8Array.BYTES_PER_ELEMENT) {
+                constructor = Uint8Array;
+            } else if (sizeInBytes === Uint16Array.BYTES_PER_ELEMENT) {
+                constructor = Uint16Array;
+            } else if (sizeInBytes === Float32Array.BYTES_PER_ELEMENT && pixelDatatype === PixelDatatype.FLOAT) {
+                constructor = Float32Array;
+            } else {
+                constructor = Uint32Array;
+            }
+
+            var size = PixelFormat.componentsLength(pixelFormat) * width * height;
+            return new constructor(size);
+        },
+
+        /**
+         * @private
+         */
+        flipY : function(bufferView, pixelFormat, pixelDatatype, width, height) {
+            var flipped = PixelFormat.createTypedArray(pixelFormat, pixelDatatype, width, height);
+            var numberOfComponents = PixelFormat.componentsLength(pixelFormat);
+            var textureWidth = width * numberOfComponents;
+            for (var i = 0; i < height; ++i) {
+                for (var j = 0; j < textureWidth; ++j) {
+                    flipped[(height - i - 1) * height + j] = bufferView[i * height + j];
```

Comment:
Why not factor `(height - i - 1) * height` and `i * height` out of the inter loop?